sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/source/KinesisRecordCoder.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.kinesis.source;
+
+import org.apache.beam.sdk.coders.ByteArrayCoder;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.coders.CoderException;
+import org.apache.beam.sdk.coders.InstantCoder;
+import org.apache.beam.sdk.coders.StandardCoder;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.coders.VarLongCoder;
+import org.apache.beam.sdk.io.kinesis.client.response.KinesisRecord;
+
+import com.amazonaws.services.kinesis.model.Record;
+import org.joda.time.Instant;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Created by p.pastuszka on 07/04/16.
+ */
+public class KinesisRecordCoder extends StandardCoder<KinesisRecord> {
```

Comment:
Use `CoderProperties` to test. See, e.g., [`MapCoderTest`](https://github.com/apache/incubator-beam/blob/69e22023f7efe3a216776297293e9d9c34d8e534/sdks/java/core/src/test/java/org/apache/beam/sdk/coders/MapCoderTest.java)